benchmark/Streamly/Benchmark/Data/Array.hs

```diff
@@ -66,6 +70,26 @@ benchIO' name src f = bench name $ nfIO $
 benchIOSink :: NFData b => Int -> String -> (Ops.Stream Int -> IO b) -> Benchmark
 benchIOSink value name f = benchIO' name (Ops.sourceIntFromTo value) f
 
+{-# INLINE sourceUnfoldrM #-}
+sourceUnfoldrM :: (S.IsStream t, S.MonadAsync m) => Int -> Int -> t m Int
+sourceUnfoldrM value n = S.unfoldrM step n
+    where
+    step cnt =
+        if cnt > n + value
+        then return Nothing
+        else return (Just (cnt, cnt + 1))
+
+{-# INLINE source #-}
+source :: (S.MonadAsync m, S.IsStream t) => Int -> Int -> t m Int
+source = sourceUnfoldrM
```

Comment:
`sourceUnfoldrM` is available in `lib/Streamly/Benchmark/Prelude.hs`, we can use the lib.